lib/class-wp-patterns-registry.php

```diff
@@ -37,7 +37,10 @@ public function register( $pattern_name, $pattern_properties ) {
 			return false;
 		}
 
-		$this->registered_patterns[ $pattern_name ] = $pattern_properties;
+		$this->registered_patterns[ $pattern_name ] = array_merge(
+			$pattern_properties,
+			array( 'name' => $pattern_name )
```

Comment:
I assume from line 17 we're using long array syntax for backwards compat so that's why I'm using it here.

Reply:
I honestly don't know whether we can use the new syntax or not 🤷